builder/azure/common/devicelogin.go

```diff
@@ -187,12 +199,11 @@ func mkTokenCallback(path string) adal.TokenRefreshCallback {
 // sure if the access_token valid, if not it uses SDK’s functionality to
 // automatically refresh the token using refresh_token (which might have
 // expired). This check is essentially to make sure refresh_token is good.
-func validateToken(env azure.Environment, token *adal.ServicePrincipalToken) error {
-	c := subscriptions.NewClientWithBaseURI(env.ResourceManagerEndpoint)
-	c.Authorizer = autorest.NewBearerAuthorizer(token)
-	_, err := c.List(context.TODO())
+func validateToken(env string, token *adal.ServicePrincipalToken) error {
+	err := token.EnsureFresh()
```

Comment:
This doesn't validate that the token can do anything for useful... not that the previous test was super useful.
If we are going here, then I would just remove this whole method.

Reply:
fixed and removed the validate method in favor for EnsureFresh